workers/generic-worker/insecure.go

```diff
@@ -26,17 +27,35 @@ func secure(configFile string) {
 	log.Printf("WARNING: can't secure generic-worker config file %q", configFile)
 }
 
-func (task *TaskRun) generateInteractiveCommand(ctx context.Context) (*exec.Cmd, error) {
+func (task *TaskRun) generateInteractiveCommand(d2gConversionInfo *d2g.ConversionInfo, ctx context.Context) (*exec.Cmd, error) {
+	var cmd []string
+
+	if d2gConversionInfo != nil {
+		cmd = []string{d2gConversionInfo.ContainerEngine, "exec", "-it", d2gConversionInfo.ContainerName, "/bin/bash"}
+	} else {
+		cmd = []string{"bash"}
+	}
+
+	return task.newCommandForInteractive(cmd, ctx)
+}
+
+func (task *TaskRun) generateInteractiveIsReadyCommand(d2gConversionInfo *d2g.ConversionInfo, ctx context.Context) (*exec.Cmd, error) {
+	cmd := []string{"/bin/bash", "-cx", "/bin/[ \"`/usr/bin/" + d2gConversionInfo.ContainerEngine + " container inspect -f '{{.State.Running}}' " + d2gConversionInfo.ContainerName + "`\" = \"true\" ]"}
```

Comment:
Are we sure podman/docker are always in `/usr/bin` ? If we want to enforce this, we should probably update docs. Isn't it sufficient that they are in `PATH`? System tools like `rm` we tend to specify absolute paths for, but I wonder if different linux distros will put them in different places, or if some workers might want to use a locally built version of podman/docker etc.

Reply:
Yeah so the issue here is that there are 2 sets of environment variables available to us. The task one which the user would be able to modify. (and `PATH="/path/to/where/my/mounts/are:..."` means they could run that interactively and bypass being dropped in the container.
The other set of env variable available at this point are the workers which might contain sensitive data?

We could just create our own set with just `PATH=/bin:/usr/bin:/sbin/usr/sbin` if you prefer but that doesn't help with non standard installation paths.

Reply:
I guess we could take the worker environment and filter everything but the `PATH` out 🤔

Reply:
I think `/bin/bash` is launched in a subshell of generic-worker, and will inherit the PATH of the parent process, which should contain the docker/podman parent directory. Is it not sufficient to just launch `d2gConversionInfo.ContainerEngine` as a command, does the full path need to be specified? My assumption is that the worker process will be configured with a PATH containing all the tools it needs.

Reply:
I'll try getting the path from the worker process and ignoring everything else in the env. This sounds like the safest thing